src/Core/CLanguage/NamedDataTypeExtractor.cs

```diff
@@ -254,9 +254,9 @@ public SerializedKind GetArgumentKindFromAttributes(string paramType, List<CAttr
                     // We have a reko::arg(register, prefix; get the register.
                     if (attr.Tokens.Count < 1 || attr.Tokens[2].Type != CTokenType.StringLiteral)
                         throw new FormatException("[[reko::arg(register,<name>)]] attribute expects a register name.");
-                    kind = new Register_v1 { Name = (string)attr.Tokens[2].Value };
+                    kind = new Register_v1 { Name = (string)attr.Tokens[2].Value! };
                 } else if (attr.Tokens[0].Type == CTokenType.Id &&
-                           (string)attr.Tokens[0].Value == "fpu")
+                           (string)attr.Tokens[0].Value! == "fpu")
```

Comment:
In this case, because the `CTokenType` is `Id`, we are guaranteed by the `CLexer` that `Value` is non-null. 

I understand the concern for the null-forgiveness operator here; it's potentially dangerous if used carelessly. 

A possible way around this is to rewrite the `CLexer` so that instead of returning a single type `CToken`, it instead returns subclasses of `CToken`, like `CIdToken`, `CNumericToken` (both of which would have an appropriate `Value`); and `CSimpleToken` (which has no `Value).
```C#
public abstract class CToken {     // Has no Value property

    public CToken(CTokenType type) { Type = type; }
}

public class CIDToken : CToken { 
    
    public CIdToken(CTokenType type, string value) : base(type)
    {
        Value = value;
    }

    public string Value {get;} // non-nullable
}
```

I haven't done so and taken this more "dangerous" approach since I'm concerned that all the required statements:
```C#
if (token is CIdToken idToken) {
   DoSomething(idToken.Value);     // Value is guaranteed to have a non-null value.
```
which involve .NET runtime casts are going to be slower than the cheaper (current):
```C#
if (token.Type == CTokenType.Id) {
    DoSomething((string)token.Value!);    // Value is guaranteed by CLexer to have a non-null value
```
and we are parsing large C header files (like `windows.h`). Especially the many switch statements in CParser.cs would be affected. A switch statement can be very efficiently implemented when the expression is an enum (typically a simple table lookup), but if the switch expression is used in "pattern matching" mode, you are not only doing type identity checks + casts, but the switch necessarily has to be translated into a series of `if` statements, which will be slower than a table lookup.

So here, for performance reasons, I go with the unsafer approach.